test/client.test.ts

```diff
@@ -3,20 +3,43 @@ import * as sinon from 'sinon';
 
 import {
   Etcd3,
+  EtcdAuthenticationFailedError,
   EtcdLeaseInvalidError,
   EtcdLockFailedError,
+  EtcdPermissionDeniedError,
+  EtcdRoleExistsError,
+  EtcdRoleNotFoundError,
+  EtcdRoleNotGrantedError,
+  EtcdUserExistsError,
+  EtcdUserNotFoundError,
   GRPCConnectFailedError,
   Lease,
+  Role,
 } from '../src';
-import { getHosts } from './util';
+import { getOptions } from './util';
+
+function expectReject(promise: Promise<any>, err: { new (message: string): Error }) {
+  return promise
+    .then(() => { throw new Error('expected to reject'); })
+    .catch(actualErr => {
+      if (!(actualErr instanceof err)) {
+        console.error(actualErr.stack);
```

Comment:
I actually do intend this here. The Chai instanceof check failing only prints the error message, it's useful to print the stack too to help diagnose issues (similarly to how we dump responses entirely on failed assertions in the backend integration tests).